libs/sdk-ui-ext/src/internal/dashboardEmbedding/KpiContent/KpiValue.tsx

```diff
@@ -44,15 +77,32 @@ class KpiValue extends PureComponent<IKpiValueProps & WrappedComponentProps> {
     }
 
     renderValue() {
-        if (this.props.isLoading) {
+        const { isLoading, error, disableKpiDrillUnderline, enableCompactSize, clientHeight } = this.props;
+        if (isLoading) {
             return <LoadingDots className={"kpi-value-loading gd-loading-dots-centered"} />;
         }
 
-        const value = this.props.error ? this.formatMessage("error") : this.renderFormattedValue();
+        const value = error ? this.formatMessage("error") : this.renderFormattedValue();
         const valueClassNames = cx("kpi-value-value", "s-kpi-value", {
-            "kpi-link-style-underline": !this.props.disableKpiDrillUnderline,
+            "kpi-link-style-underline": !disableKpiDrillUnderline,
         });
 
+        if (enableCompactSize) {
+            if (!clientHeight) {
+                return <LoadingDots className={"kpi-value-loading gd-loading-dots-centered"} />;
```

Comment:
Please add a comment why we display loading if `!clientSize`. It is not obvious.